### PR TITLE
perf(lsp): Call `serverRequest` via V8 instead of via `executeScript`

### PR DESCRIPTION
**Pull request by nathanwhit**
Doesn't have a noticeable perf impact from my benchmarking, but theoretically should be better.

### CODE REVIEW

File: cli/lsp/tsc.rs [hunk 1]
```diff
@@ -5236,7 +5279,7 @@ mod tests {
   }
 
   #[tokio::test]
-  async fn test_completions() {
+  async fn test_completionss() {
```
Comment (mmastrac) [inline cli/lsp/tsc.rs:5282]: Accidental?

Comment (nathanwhit) [reply cli/lsp/tsc.rs:5282]: oh good catch, didn't mean to commit that. 

It's kinda dumb, but sometimes when I want to run only a single test and the test happens to be a prefix of another test, i just add some chars to the end to make it unique. Forgot to revert it before staging the commit😅

**Comment by bartlomieju** [comment]
Re-posting for visibility: maybe we could have a giant enum that you send over the channel and perform direct v8 serialization in the TSC thread?

File: cli/lsp/tsc.rs [hunk 2]
```diff
@@ -4623,57 +4631,286 @@ pub struct GetNavigateToItemsArgs {
...
+  /// Converts the request into a tuple containing the method name and the
+  /// arguments (in the form of a V8 value) to be passed to the server request
+  /// function
+  fn to_server_request<'s>(
+    &self,
+    scope: &mut v8::HandleScope<'s>,
+  ) -> Result<(&'static str, Option<v8::Local<'s, v8::Value>>), AnyError> {
+    let args = match self {
+      TscRequest::ProjectChanged(args) => {
+        ("$projectChanged", Some(serde_v8::to_v8(scope, args)?))
```
Comment (bartlomieju) [inline cli/lsp/tsc.rs:4774]: Suggestion: consider giving these a numeric ID that you then map back in `99_compiler_main.js` - it's gonna be faster than having to serialize these strings, but it's gonna be a bit wonky to keep track of them. Might be worth the trade-off 🤞

**Review by bartlomieju — APPROVED** [review]
Looks good to me, I would still encourage to try and make these reqs have numeric IDs instead of sending strings over.

File: cli/lsp/tsc.rs (hunk 2 above)
Comment (mmastrac) [reply cli/lsp/tsc.rs:4774]: We could do something like this to make it more robust to maintain a numeric mapping:

1. Use `enum_derive` or something like it to derive an iterator for all variants on the Rust side.
2. Keep an enum in JS with the numeric mappings in some sort of object form (ie: easily converted to a string with the names and values):

```
enum TscRequest {
  requestA = 0,
  requestB,
  ...
}
```

4. Have a callback that requests the full contents of that enum, only used for a unit test in the next step
5. Have a unit test on the CLI/LSP side that compares whether the JS enum and Rust enums is up-to-date or not.

**Pull request merged**